dask_expr/_reductions.py

```diff
@@ -65,58 +85,140 @@ def _layer(self):
+            try:
+                if v != self.kind._defaults[k]:
+                    header += f" {k}={v}"
+            except KeyError:
+                header += f" {k}={v}"
+
+        lines = [header] + lines
+        lines = [" " * indent + line for line in lines]
+        return lines
+
+
+class TreeReduce(Expr):
+    """Tree-reduction component of `ApplyConcatApply`
+
+    This class is used within `ApplyConcatApply._lower`.
+
+    See Also
+    --------
+    ApplyConcatApply
+    """
+
+    _parameters = [
+        "frame",
+        "kind",
+        "_meta",
+        "combine",
+        "aggregate",
+        "combine_kwargs",
+        "aggregate_kwargs",
+    ]
+
+    def __dask_postcompute__(self):
+        return toolz.first, ()
+
+    def _layer(self):
         # apply combine to batches of intermediate results
+        j = 1
+        d = {}
+        keys = self.frame.__dask_keys__()
+        split_every = getattr(self, "split_every", 0)
         while len(keys) > 1:
             new_keys = []
             for i, batch in enumerate(
                 toolz.partition_all(split_every or len(keys), keys)
             ):
                 batch = list(batch)
-                if combine_kwargs:
-                    d[self._name, j, i] = (apply, combine, [batch], combine_kwargs)
+                if self.combine_kwargs:
+                    d[self._name, j, i] = (
+                        apply,
+                        self.combine,
+                        [batch],
+                        self.combine_kwargs,
+                    )
                 else:
-                    d[self._name, j, i] = (combine, batch)
+                    d[self._name, j, i] = (self.combine, batch)
                 new_keys.append((self._name, j, i))
             j += 1
             keys = new_keys
 
         # apply aggregate to the final result
-        d[self._name, 0] = (apply, aggregate, [keys], aggregate_kwargs)
+        d[self._name, 0] = (apply, self.aggregate, [keys], self.aggregate_kwargs)
 
         return d
 
     @property
     def _meta(self):
-        meta = meta_nonempty(self.frame._meta)
-        meta = self.chunk(meta, **self.chunk_kwargs)
-        aggregate = self.aggregate or (lambda x: x)
-        if self.combine:
-            combine = self.combine
-            combine_kwargs = self.combine_kwargs
-        else:
-            combine = aggregate
-            combine_kwargs = self.aggregate_kwargs
-
-        meta = combine([meta], **combine_kwargs)
-        meta = aggregate([meta], **self.aggregate_kwargs)
-        return make_meta(meta)
+        return self.operand("_meta")
 
     def _divisions(self):
         return (None, None)
 
+    def __str__(self):
+        chunked = str(self.frame)
+        split_every = getattr(self, "split_every", 0)
+        return f"{type(self).__name__}({chunked}, kind={funcname(self.kind)}, split_every={split_every})"
+
+    def _tree_repr_lines(self, indent=0, recursive=True):
```

Comment:
Does it make sense to add tests for both ``_tree_repr_lines``?

Reply:
Hmm. I do agree that we don't have the best coverage here, but I'm also a bit uncomfortable with the "fragility" of testing specific details of the `pprint` output, or private `_tree_repr_lines` results.

That said, I'll be happy to revise if you have something in mind.

Reply:
I was more concerned that we don't seem to hit this at all (I might be wrong). Having a simple test that checks that this renders at all would be sufficient for now. But not urgent, this isn't critical

Reply:
[test_tree_repr](https://github.com/phofl/dask-expr/blob/0f577c6829472f9c02a655998fcf64202ee27886/dask_expr/tests/test_collection.py#L578) should be checking that these methods are producing `"Sum(Chunk):"` and `"Sum(TreeReduce): split_every=0"` for the `Chunk` and `TreeReduce` cases, respectively.